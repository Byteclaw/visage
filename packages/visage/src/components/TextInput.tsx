import React, {
  forwardRef,
  ReactElement,
  useCallback,
  useState,
  Ref,
  FocusEventHandler,
} from 'react';
import {
  markAsVisageComponent,
  ExtractVisageComponentProps,
  VisageComponent,
} from '@byteclaw/visage-core';
import { createComponent } from '../core';
import {
  createControlFocusShadow,
  disabledControlStyles,
  disabledControlBooleanVariant,
  invalidControlStyles,
  invalidControlBooleanVariant,
} from './shared';

export const TextInputBaseStyles: VisageStyleSheet = {
  backgroundColor: 'color(shades tint(10%))',
  borderColor: 'color(shades shade(30%))',
  borderStyle: 'solid',
  borderWidth: '1px',
  borderRadius: 'controlBorderRadius',
  fontFamily: 'inherit',
  fontSize: 'inherit',
  lineHeight: 'inherit',
  outline: 'none',
  m: 0,
  p: 0,
  position: 'relative',
  // data-focused is used by text input on base
  '&:focus, &[data-focused="true"]': {
    boxShadow: createControlFocusShadow('accent'),
  },
};

const InputExtraElement = createComponent('div', {
  displayName: 'TextInputExtraElement',
  styles: {
    color: 'shadesText',
    display: 'flex',
    flexBasis: 'auto',
    flexGrow: 0,
    flexShrink: 0,
    fontSize: 'inherit',
    lineHeight: 'inherit',
    maxWidth: '100%',
    position: 'relative',
    userSelect: 'none',
    px: 1,
  },
});

const TextInputControlBase = createComponent('div', {
  displayName: 'TextInputControlBase',
  styles: {
    display: 'flex',
    flexGrow: 1,
    flexShrink: 1,
    flexBasis: '0%',
    flexWrap: 'wrap',
    width: '100%',
  },
});

const TextInputControl = createComponent('input', {
  displayName: 'TextInputControl',
  styles: {
    appearance: 'none',
    backgroundColor: 'transparent',
    border: 'none',
    color: 'shadesText',
    cursor: 'inherit',
    flexGrow: 1,
    flexShrink: 1,
    flexBasis: '0%',
    fontFamily: 'inherit',
    fontSize: 'inherit',
    lineHeight: 'inherit',
    outline: 'none',
    m: 0,
    py: 1,
    px: 1,
    width: '100%',
    '&[data-prefix]': {
      pl: 0,
    },
    '&[data-suffix]': {
      pr: 0,
    },
    '::placeholder': {
      color: 'color(shadesText alpha(0.3))',
    },
  },
});

const InputBase = createComponent('div', {
  displayName: 'TextInputBase',
  styles: props => ({
    ...TextInputBaseStyles,
    alignItems: 'center',
    display: 'flex',
    flexWrap: 'wrap',
    ...(props.disabled ? disabledControlStyles : {}),
    ...(props.invalid ? invalidControlStyles : {}),
  }),
  variants: [disabledControlBooleanVariant, invalidControlBooleanVariant],
});

type InputProps = Omit<JSX.IntrinsicElements['input'], 'prefix'>;

interface TextInputProps {
  baseProps?: ExtractVisageComponentProps<typeof InputBase>;
  invalid?: boolean;
  prefix?: ReactElement;
  prefixProps?: ExtractVisageComponentProps<typeof InputExtraElement>;
  /** Elements included between prefix and input */
  prefixExtra?: ReactElement;
  suffix?: ReactElement;
  /** Elements included between input and suffix */
  suffixExtra?: ReactElement;
  suffixProps?: ExtractVisageComponentProps<typeof InputExtraElement>;
}

export const TextInput: VisageComponent<
  TextInputProps & InputProps
> = forwardRef(
  (
    {
      baseProps,
      disabled,
      invalid,
      onBlur: outerOnBlur,
      onFocus: outerOnFocus,
      prefix,
      prefixExtra,
      prefixProps,
      suffix,
      suffixExtra,
      suffixProps,
      ...restProps
    }: TextInputProps & InputProps,
    ref: Ref<HTMLInputElement>,
  ) => {
    const [focused, setFocused] = useState(false);
    const onBlur: FocusEventHandler<HTMLInputElement> = useCallback(
      e => {
        setFocused(false);

        if (outerOnBlur) outerOnBlur(e);
      },
      [outerOnBlur],
    );
    const onFocus: FocusEventHandler<HTMLInputElement> = useCallback(
      e => {
        setFocused(true);

        if (outerOnFocus) outerOnFocus(e);
      },
      [outerOnFocus],
    );

    return (
      <InputBase
        {...baseProps}
        data-focused={focused}
        disabled={disabled}
        invalid={invalid}
      >
        {prefix ? (
          <InputExtraElement {...prefixProps} data-prefix>
            {prefix}
          </InputExtraElement>
        ) : null}
        <TextInputControlBase>
          {prefixExtra}
          <TextInputControl
            aria-invalid={invalid}
            data-prefix={prefix ? true : undefined}
            data-suffix={suffix ? true : undefined}
            disabled={disabled}
            onBlur={onBlur}
            onFocus={onFocus}
            ref={ref}
            {...restProps}
          />
          {suffixExtra}
        </TextInputControlBase>
        {suffix ? (
          <InputExtraElement {...suffixProps} data-suffix>
            {suffix}
          </InputExtraElement>
        ) : null}
      </InputBase>
    );
  },
) as any;

markAsVisageComponent(TextInput);
