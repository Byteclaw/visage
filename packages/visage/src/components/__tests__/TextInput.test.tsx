import React from 'react';
import { render } from './render';
import { TextInput } from '../TextInput';

describe('TextInput', () => {
  it('render correctly', () => {
    const { asFragment } = render(<TextInput />);

    expect(asFragment()).toMatchInlineSnapshot(`
      <DocumentFragment>
        .emotion-0 {
        -webkit-box-pack: center;
        -webkit-justify-content: center;
        -ms-flex-pack: center;
        justify-content: center;
        display: -webkit-box;
        display: -webkit-flex;
        display: -ms-flexbox;
        display: flex;
        position: fixed;
        bottom: 0;
        left: 0;
        right: 0;
        z-index: 600;
      }

      <div
          aria-live="polite"
          class="emotion-0"
          data-toastmanager="true"
        />
        .emotion-2 {
        background-color: textInput;
        border-color: textInputBorder;
        border-style: solid;
        border-width: 1px;
        border-radius: 4px;
        font-family: inherit;
        font-size: inherit;
        line-height: inherit;
        outline: none;
        margin: 0px;
        padding: 0px;
        position: relative;
        -webkit-align-items: center;
        -webkit-box-align: center;
        -ms-flex-align: center;
        align-items: center;
        display: -webkit-box;
        display: -webkit-flex;
        display: -ms-flexbox;
        display: flex;
        -webkit-flex-wrap: wrap;
        -ms-flex-wrap: wrap;
        flex-wrap: wrap;
      }

      .emotion-2:focus,
      .emotion-2[data-focused="true"] {
        box-shadow: 0 0 0 4px rgba(255,255,255,0.4),0 0 0 4px accent;
      }

      .emotion-1 {
        display: -webkit-box;
        display: -webkit-flex;
        display: -ms-flexbox;
        display: flex;
        -webkit-box-flex: 1;
        -webkit-flex-grow: 1;
        -ms-flex-positive: 1;
        flex-grow: 1;
        -webkit-flex-shrink: 1;
        -ms-flex-negative: 1;
        flex-shrink: 1;
        -webkit-flex-basis: 0%;
        -ms-flex-preferred-size: 0%;
        flex-basis: 0%;
        -webkit-flex-wrap: wrap;
        -ms-flex-wrap: wrap;
        flex-wrap: wrap;
        width: 100%;
      }

      .emotion-0 {
        -webkit-appearance: none;
        -moz-appearance: none;
        appearance: none;
        background-color: transparent;
        border: none;
        color: currentColor;
        cursor: inherit;
        -webkit-box-flex: 1;
        -webkit-flex-grow: 1;
        -ms-flex-positive: 1;
        flex-grow: 1;
        -webkit-flex-shrink: 1;
        -ms-flex-negative: 1;
        flex-shrink: 1;
        -webkit-flex-basis: 0%;
        -ms-flex-preferred-size: 0%;
        flex-basis: 0%;
        font-family: inherit;
        font-size: inherit;
        line-height: inherit;
        outline: none;
        margin: 0px;
        padding-top: 8px;
        padding-bottom: 8px;
        padding-left: 8px;
        padding-right: 8px;
        width: 100%;
      }

      .emotion-0[data-prefix] {
        padding-left: 0px;
      }

      .emotion-0[data-suffix] {
        padding-right: 0px;
      }

      .emotion-0::-webkit-input-placeholder {
        color: currentColor;
        opacity: 0.3;
      }

      .emotion-0::-moz-placeholder {
        color: currentColor;
        opacity: 0.3;
      }

      .emotion-0:-ms-input-placeholder {
        color: currentColor;
        opacity: 0.3;
      }

      .emotion-0::placeholder {
        color: currentColor;
        opacity: 0.3;
      }

      <div
          class="emotion-2"
          data-disabled="false"
          data-focused="false"
          data-invalid="false"
        >
          <div
            class="emotion-1"
          >
            <input
              class="emotion-0"
            />
          </div>
        </div>
      </DocumentFragment>
    `);
  });
});
