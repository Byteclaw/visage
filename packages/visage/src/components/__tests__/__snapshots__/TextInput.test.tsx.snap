// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput is extendable by InputExtraElement face 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  .emotion-3 {
  background-color: rgb(255,255,255);
  border-color: rgb(0,0,0);
  border-style: solid;
  border-width: 1px;
  border-radius: 4px;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  outline: none;
  margin: 0px;
  padding: 0px;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-3:focus,
.emotion-3[data-focused="true"] {
  box-shadow: 0 0 0 4px rgba(255,255,255,0.4),0 0 0 4px rgb(0,0,255);
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 0%;
  -ms-flex-preferred-size: 0%;
  flex-basis: 0%;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
}

.emotion-1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: rgba(0,0,0,0);
  border: none;
  color: rgb(255,255,255);
  cursor: inherit;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 0%;
  -ms-flex-preferred-size: 0%;
  flex-basis: 0%;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  outline: none;
  margin: 0px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 8px;
  padding-right: 8px;
  width: 100%;
}

.emotion-1[data-prefix] {
  padding-left: 0px;
}

.emotion-1[data-suffix] {
  padding-right: 0px;
}

.emotion-1::-webkit-input-placeholder {
  color: rgba(255,255,255,0.3);
}

.emotion-1::-moz-placeholder {
  color: rgba(255,255,255,0.3);
}

.emotion-1:-ms-input-placeholder {
  color: rgba(255,255,255,0.3);
}

.emotion-1::placeholder {
  color: rgba(255,255,255,0.3);
}

.emotion-0 {
  color: rgb(255,255,255);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-size: inherit;
  line-height: inherit;
  max-width: 100%;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding-left: 8px;
  padding-right: 8px;
  border-radius: 10px;
}

.emotion-0:hover {
  color: rgb(255,192,203);
}

<div
    class="emotion-3"
    data-disabled="false"
    data-focused="false"
    data-invalid="false"
  >
    <div
      class="emotion-0"
      data-prefix="true"
    >
      <span>
        P
      </span>
    </div>
    <div
      class="emotion-2"
    >
      <input
        class="emotion-1"
        data-prefix="true"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextInput is extendable by TextInputBase face 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  .emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 0%;
  -ms-flex-preferred-size: 0%;
  flex-basis: 0%;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
}

.emotion-0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: rgba(0,0,0,0);
  border: none;
  color: rgb(255,255,255);
  cursor: inherit;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 0%;
  -ms-flex-preferred-size: 0%;
  flex-basis: 0%;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  outline: none;
  margin: 0px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 8px;
  padding-right: 8px;
  width: 100%;
}

.emotion-0[data-prefix] {
  padding-left: 0px;
}

.emotion-0[data-suffix] {
  padding-right: 0px;
}

.emotion-0::-webkit-input-placeholder {
  color: rgba(255,255,255,0.3);
}

.emotion-0::-moz-placeholder {
  color: rgba(255,255,255,0.3);
}

.emotion-0:-ms-input-placeholder {
  color: rgba(255,255,255,0.3);
}

.emotion-0::placeholder {
  color: rgba(255,255,255,0.3);
}

.emotion-2 {
  background-color: rgb(255,255,255);
  border-color: rgb(0,0,0);
  border-style: solid;
  border-width: 1px;
  border-radius: 10px;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  outline: none;
  margin: 0px;
  padding: 0px;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-2:focus,
.emotion-2[data-focused="true"] {
  box-shadow: 0 0 0 4px rgba(255,255,255,0.4),0 0 0 4px rgb(0,0,255);
}

<div
    class="emotion-2"
    data-disabled="false"
    data-focused="false"
    data-invalid="false"
  >
    <div
      class="emotion-1"
    >
      <input
        class="emotion-0"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextInput is extendable by TextInputControl face 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  .emotion-2 {
  background-color: rgb(255,255,255);
  border-color: rgb(0,0,0);
  border-style: solid;
  border-width: 1px;
  border-radius: 4px;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  outline: none;
  margin: 0px;
  padding: 0px;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-2:focus,
.emotion-2[data-focused="true"] {
  box-shadow: 0 0 0 4px rgba(255,255,255,0.4),0 0 0 4px rgb(0,0,255);
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 0%;
  -ms-flex-preferred-size: 0%;
  flex-basis: 0%;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
}

.emotion-0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: rgba(0,0,0,0);
  border: none;
  color: rgb(255,255,255);
  cursor: inherit;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 0%;
  -ms-flex-preferred-size: 0%;
  flex-basis: 0%;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  outline: none;
  margin: 0px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 8px;
  padding-right: 8px;
  width: 100%;
  border-radius: 10px;
}

.emotion-0[data-prefix] {
  padding-left: 0px;
}

.emotion-0[data-suffix] {
  padding-right: 0px;
}

.emotion-0::-webkit-input-placeholder {
  color: rgb(255,192,203);
  opacity: 1;
}

.emotion-0::-moz-placeholder {
  color: rgb(255,192,203);
  opacity: 1;
}

.emotion-0:-ms-input-placeholder {
  color: rgb(255,192,203);
  opacity: 1;
}

.emotion-0::placeholder {
  color: rgb(255,192,203);
  opacity: 1;
}

<div
    class="emotion-2"
    data-disabled="false"
    data-focused="false"
    data-invalid="false"
  >
    <div
      class="emotion-1"
    >
      <input
        class="emotion-0"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextInput is extendable by TextInputControlBase face 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  .emotion-2 {
  background-color: rgb(255,255,255);
  border-color: rgb(0,0,0);
  border-style: solid;
  border-width: 1px;
  border-radius: 4px;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  outline: none;
  margin: 0px;
  padding: 0px;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-2:focus,
.emotion-2[data-focused="true"] {
  box-shadow: 0 0 0 4px rgba(255,255,255,0.4),0 0 0 4px rgb(0,0,255);
}

.emotion-0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: rgba(0,0,0,0);
  border: none;
  color: rgb(255,255,255);
  cursor: inherit;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 0%;
  -ms-flex-preferred-size: 0%;
  flex-basis: 0%;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  outline: none;
  margin: 0px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 8px;
  padding-right: 8px;
  width: 100%;
}

.emotion-0[data-prefix] {
  padding-left: 0px;
}

.emotion-0[data-suffix] {
  padding-right: 0px;
}

.emotion-0::-webkit-input-placeholder {
  color: rgba(255,255,255,0.3);
}

.emotion-0::-moz-placeholder {
  color: rgba(255,255,255,0.3);
}

.emotion-0:-ms-input-placeholder {
  color: rgba(255,255,255,0.3);
}

.emotion-0::placeholder {
  color: rgba(255,255,255,0.3);
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 0%;
  -ms-flex-preferred-size: 0%;
  flex-basis: 0%;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  border-radius: 10px;
}

<div
    class="emotion-2"
    data-disabled="false"
    data-focused="false"
    data-invalid="false"
  >
    <div
      class="emotion-1"
    >
      <input
        class="emotion-0"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextInput render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  .emotion-2 {
  background-color: rgb(255,255,255);
  border-color: rgb(0,0,0);
  border-style: solid;
  border-width: 1px;
  border-radius: 4px;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  outline: none;
  margin: 0px;
  padding: 0px;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-2:focus,
.emotion-2[data-focused="true"] {
  box-shadow: 0 0 0 4px rgba(255,255,255,0.4),0 0 0 4px rgb(0,0,255);
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 0%;
  -ms-flex-preferred-size: 0%;
  flex-basis: 0%;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
}

.emotion-0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: rgba(0,0,0,0);
  border: none;
  color: rgb(255,255,255);
  cursor: inherit;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 0%;
  -ms-flex-preferred-size: 0%;
  flex-basis: 0%;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  outline: none;
  margin: 0px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 8px;
  padding-right: 8px;
  width: 100%;
}

.emotion-0[data-prefix] {
  padding-left: 0px;
}

.emotion-0[data-suffix] {
  padding-right: 0px;
}

.emotion-0::-webkit-input-placeholder {
  color: rgba(255,255,255,0.3);
}

.emotion-0::-moz-placeholder {
  color: rgba(255,255,255,0.3);
}

.emotion-0:-ms-input-placeholder {
  color: rgba(255,255,255,0.3);
}

.emotion-0::placeholder {
  color: rgba(255,255,255,0.3);
}

<div
    class="emotion-2"
    data-disabled="false"
    data-focused="false"
    data-invalid="false"
  >
    <div
      class="emotion-1"
    >
      <input
        class="emotion-0"
      />
    </div>
  </div>
</DocumentFragment>
`;
